File: ProjectEuler/201-250/S243.cs
using System.Linq;
using System;

namespace ProjectEuler
{
    public class S243 : ISolution
    {
        public string GetAnswer()
        {
            int[] primes = Utils.GenPrimes(50).Where(l => l != 0).Select(Convert.ToInt32).ToArray();
            long c = 1;
            long d = 1;
            foreach (var p in primes)
            {
                c *= p - 1;
                d *= p;
                if (c * 94744L < d * 15499L)
                {
                    break;
                }
            }

            for (int i = 2; i <= primes.Last(); i++)
            {
                long n = d * i;
                if (c * n * 94744L < d * (n - 1) * 15499L)
                {
                    return n.ToString();
                }
            }

            return string.Empty;
        }
    }
}

File: src/201-250/S243.cc
#include "S243.h"
#include "../Utils.h"
#include <string>
#include <vector>

using namespace std;

namespace ProjectEuler
{
std::string S243::GetAnswer()
{
	auto primes = Utils::GenPrimes(50);
	int64_t c = 1;
	int64_t d = 1;
	for (auto p : primes)
	{
		c *= p - 1;
		d *= p;
		if (c * 94744L < d * 15499L)
		{
			break;
		}
	}

	for (int i = 2; i <= primes.back(); i++)
	{
		int64_t n = d * i;
		if ((__int128)c * n * 94744L < (__int128)d * (n - 1) * 15499L)
		{
			return std::to_string(n);
		}
	}

	return "";
}

} // namespace ProjectEuler
